homeassistant/bootstrap.py

```diff
@@ -29,6 +29,9 @@
 
 ERROR_LOG_FILENAME = "home-assistant.log"
 
+# How long to wait until things that run on bootstrap have to finish.
+TIMEOUT_EVENT_BOOTSTRAP = 15
```

Comment:
recorder and MQTT need a compiler and the frontend are over 25mb to download. Also, database migration can take a huge time. Not sure if that hardcoded value is really a good idea as hardcoded. I.e. the Supervisor reset this time each pip packages or stop the timer while the core is doing database migration.

Reply:
Anyway, I fix that like in past on the supervisor: https://github.com/home-assistant/supervisor/pull/1771 - just avoid that Home Assistant user end up with corrupt database

Reply:
That all are hacks, I would like to have a core that I don't need patch at runtime. I mean that would be okay in old days of the Supervisor but not today. I closed the PR and repaired a better one if we decide to patch it 🤷‍♂️ . My question is, why can a core module block the hole startup which is not possible to fix?

Reply:
It shouldn't ever happen but there is always the custom component that hasn't been reviewed that will cause this issue.

I can bump it up to 30 minutes to be a safeguard against this or take it out.

Reply:
For reference, I was inspired by this code:
https://github.com/home-assistant/core/blob/dev/homeassistant/core.py#L264-L275

I'm going to bump to 1800 seconds and update the comment unless there is strong desire to remove this.

Reply:
A database upgrade could take a lot longer than 30 minutes on a large db on slow hardware? 🤷‍♂️

Reply:
I'll bump it to 4 hours

Reply:
That may seem like a lot, but from reading the forums, I see people trying to fix their install for > 1 week so it should still help out these cases.

Reply:
Also, this isn't a hill I want to die on so if the consensus is to take it out, thats fine with me.  I'd be happy to be wrong about this being an issue as it would one less we don't have to deal with.